Tests/TSCUtilityTests/DownloaderTests.swift

```diff
@@ -72,6 +73,138 @@ class DownloaderTests: XCTestCase {
+            let destination = tmpdir.appending(component: "download")
+
+            let didStartLoadingExpectation = XCTestExpectation(description: "didStartLoading")
+            let progress50Expectation = XCTestExpectation(description: "progress50")
+            let progress100Expectation = XCTestExpectation(description: "progress100")
+            let successExpectation = XCTestExpectation(description: "success")
+            MockAuthenticatingURLProtocol.notifyDidStartLoading(for: url, completion: { didStartLoadingExpectation.fulfill() })
+
+            downloader.downloadFile(at: url, to: destination, withAuthorizationProvider: netrc, progress: { bytesDownloaded, totalBytesToDownload in
+                
+                XCTAssertEqual(MockAuthenticatingURLProtocol.authenticationHeader(for: url), testAuthHeader)
+
+                switch (bytesDownloaded, totalBytesToDownload) {
+                case (512, 1024):
+                    progress50Expectation.fulfill()
+                case (1024, 1024):
+                    progress100Expectation.fulfill()
+                default:
+                    XCTFail("unexpected progress")
+                }
+            }, completion: { result in
+                switch result {
+                case .success:
+                    XCTAssert(localFileSystem.exists(destination))
+                    let bytes = ByteString(Array(repeating: 0xbe, count: 512) + Array(repeating: 0xef, count: 512))
+                    XCTAssertEqual(try! localFileSystem.readFileContents(destination), bytes)
+                    successExpectation.fulfill()
+                case .failure(let error):
+                    XCTFail("\(error)")
+                }
+            })
+
+            wait(for: [didStartLoadingExpectation], timeout: 1.0)
+
+            let response = HTTPURLResponse(url: url, statusCode: 200, httpVersion: "1.1", headerFields: [
+                "Content-Length": "1024"
+            ])!
+
+            MockAuthenticatingURLProtocol.sendResponse(response, for: url)
+            MockAuthenticatingURLProtocol.sendData(Data(repeating: 0xbe, count: 512), for: url)
+            wait(for: [progress50Expectation], timeout: 1.0)
+            MockAuthenticatingURLProtocol.sendData(Data(repeating: 0xef, count: 512), for: url)
+            wait(for: [progress100Expectation], timeout: 1.0)
+            MockAuthenticatingURLProtocol.sendCompletion(for: url)
+            wait(for: [successExpectation], timeout: 1.0)
+        }
+      #endif
+    }
+    
+    @available(OSX 10.13, *)
+    func testDefaultAuthenticationSuccess() {
+
+        let netrcContent = "default login default password default"
+        guard case .success(let netrc) = Netrc.from(netrcContent) else {
+            return XCTFail("Cannot load netrc content")
+        }
+        let authData = "default:default".data(using: .utf8)!
+        let testAuthHeader = "Basic \(authData.base64EncodedString())"
+        
```

Comment:
If the rest of the test case is not executed on anything other than macOS, is this setup code needed?

While we're at it, are there any strong reasons for not testing this on Linux?

Reply:
@MaxDesiatov The platform constraint is to guarantee the availability of `NSTextCheckingResult.range(withName name: String) -> NSRange`.  To be honest, I don't know enough about the linux port to say whether this is supportable or not.  In the corelibs project, the method is unimplemented [here](https://github.com/apple/swift-corelibs-foundation/blob/2a5bc4d8a0b073532e60410682f5eb8f00144870/Sources/Foundation/NSTextCheckingResult.swift#L71).

Reply:
I understand, my point here is that the test function doesn't test anything on Linux, the setup code is meaningless. Maybe there could be some other way to implement the test without that function? Otherwise, could you guard the whole test to supported platforms and add a comment that clarifies the reason for it?

Reply:
@MaxDesiatov I think this may already be there.  Can you confirm if [this](https://github.com/apple/swift-tools-support-core/blob/b44e6c43d2e1ef89aa0322115d82f06943187a6e/Tests/TSCUtilityTests/DownloaderTests.swift#L77) is what you mean? Or do you prefer `#if os(macOS)` style?

Reply:
The star in the `@available` condition means that it's available on all other platforms. I'm not sure if `@available(Linux, unavailable)` would work, there's [an old post on Forums](https://forums.swift.org/t/updates-to-the-available-attribute/10951) that reports it didn't in 2018. If it still doesn't, I think the whole test function should be guarded with `#if os(macOS)` with a corresponding comment that explains the reason for that.

Reply:
@MaxDesiatov decided to wrap in `#if os(macOS)`.  Will hop over to spm PR to replicate.

Reply:
This eliminates the requirement for swift-driver to bump min platform to 10.13.